doc/design/externally-manage-pf.md

```diff
@@ -0,0 +1,174 @@
+flow of virtual functions *SKIPPING* only the Virtual function allocation.
+
+The SR-IOV operator will update the SR-IOV Network Device Plugin with the pool information
+
+Another change with the operator beavior is when we delete a policy with had `externallyManaged: true` the SR-IOV operator
+will *NOT* reset the `numVfs`
+
+### API Extensions
+
+For SriovNetworkNodePolicy
+
+```golang
+// SriovNetworkNodePolicySpec defines the desired state of SriovNetworkNodePolicy
+type SriovNetworkNodePolicySpec struct {
+// SRIOV Network device plugin endpoint resource name
+ResourceName string `json:"resourceName"`
+// NodeSelector selects the nodes to be configured
+NodeSelector map[string]string `json:"nodeSelector"`
+// +kubebuilder:validation:Minimum=0
+// +kubebuilder:validation:Maximum=99
+// Priority of the policy, higher priority policies can override lower ones.
+Priority int `json:"priority,omitempty"`
+// +kubebuilder:validation:Minimum=1
+// MTU of VF
+Mtu int `json:"mtu,omitempty"`
+// +kubebuilder:validation:Minimum=0
+// Number of VFs for each PF
+NumVfs int `json:"numVfs"`
+// NicSelector selects the NICs to be configured
+NicSelector SriovNetworkNicSelector `json:"nicSelector"`
+// +kubebuilder:validation:Enum=netdevice;vfio-pci
+// The driver type for configured VFs. Allowed value "netdevice", "vfio-pci". Defaults to netdevice.
+DeviceType string `json:"deviceType,omitempty"`
+// RDMA mode. Defaults to false.
+IsRdma bool `json:"isRdma,omitempty"`
+// mount vhost-net device. Defaults to false.
+NeedVhostNet bool `json:"needVhostNet,omitempty"`
+// +kubebuilder:validation:Enum=eth;ETH;ib;IB
+// NIC Link Type. Allowed value "eth", "ETH", "ib", and "IB".
+LinkType string `json:"linkType,omitempty"`
+// +kubebuilder:validation:Enum=legacy;switchdev
+// NIC Device Mode. Allowed value "legacy","switchdev".
+EswitchMode string `json:"eSwitchMode,omitempty"`
+// +kubebuilder:validation:Enum=virtio
+// VDPA device type. Allowed value "virtio"
+VdpaType string `json:"vdpaType,omitempty"`
+// Exclude device's NUMA node when advertising this resource by SRIOV network device plugin. Default to false.
+ExcludeTopology bool `json:"excludeTopology,omitempty"`
++ // don't create the virtual function only assign to the driver and allocated them to device plugin. Defaults to false.
++ ExternallyManaged bool `json:"externallyManaged,omitempty"`
+}
+```
+
+For SriovNetworkNodeState
+
+```golang
+type Interface struct {
+PciAddress        string    `json:"pciAddress"`
+NumVfs            int       `json:"numVfs,omitempty"`
+Mtu               int       `json:"mtu,omitempty"`
+Name              string    `json:"name,omitempty"`
+LinkType          string    `json:"linkType,omitempty"`
+EswitchMode       string    `json:"eSwitchMode,omitempty"`
+VfGroups          []VfGroup `json:"vfGroups,omitempty"`
++ ExternallyManaged bool      `json:"externallyManaged,omitempty"`
+}
+```
+
+### Implementation Details/Notes/Constraints
```

Comment:
can we have changes that need to be applied to the different components under this section ?

- controller
- webhook
- config daemon